File: 2022/day-1/day-1.jl
open("input.txt") do file
    lines = readlines(file)

    calories_held = 0
    calories = 0
    for line in lines
        if(line != "")
            calories += parse(Int, line)
        elseif(calories > calories_held)
            calories_held = calories
            calories = 0
        else
            calories = 0
        end
    end
    if(calories > calories_held)
        calories_held = calories
    end

    println("Part 1 solution: $(calories_held)")

    calories_held = Vector{Int64}()
    calories = 0
    for line in lines
        if(line != "")
            calories += parse(Int, line)
        else
            append!(calories_held, calories)
            calories = 0
        end
    end
    append!(calories_held, calories)
    println("Part 2 solution: $(sum(first(sort!(calories_held, rev = true), 3)))")
end

File: 2022/day-1/day-1.py
with open("input.txt", "r") as f:
    lines = f.readlines()
    numbers = [line.strip() for line in lines]

def count_calories(numbers):
    calories_held = []
    calories = 0
    for i in numbers:
        if i:
            calories += int(i)
        else:
            calories_held.append(calories)
            calories = 0

    calories_held.append(calories)

    return calories_held

def part1(x):
    return max(x)

def part2(x):
    return sum(sorted(x)[-3:])

if __name__ == "__main__":
    print('Part 1 solution:', part1(count_calories(numbers)))
    print('Part 2 solution:', part2(count_calories(numbers)))
